src/constructs/iam/roles/instance-role.ts

```diff
@@ -10,30 +11,33 @@ import {
 } from "../policies";
 import { GuRole } from "./roles";
 
-interface GuInstanceRoleProps {
+interface GuInstanceRoleProps extends AppIdentity {
   withoutLogShipping?: boolean; // optional to have log shipping added by default, you have to opt out
   additionalPolicies?: GuPolicy[];
 }
 
 export class GuInstanceRole extends GuRole {
   private policies: GuPolicy[];
 
-  constructor(scope: GuStack, id: string = "InstanceRole", props?: GuInstanceRoleProps) {
-    super(scope, id, {
+  // eslint-disable-next-line custom-rules/valid-constructors -- TODO be better
```

Comment:
As we are starting to regularly break this convention perhaps we should consider relaxing the linting rule?

Reply:
Definitely! I started on this a couple of weeks ago, but looks like I deleted my branch! 😱 

I _think_ the rules should be:
- 1, 2 or 3 arguments
- 1st argument is always GuStack
- If only 2 arguments, 2nd argument is required and are `props` w/no default
- If 3 arguments, 2nd is ID w/no default, 3rd are `props` w/no default and optional

Will raise this in a separate PR to see how it pans out.

Reply:
Agree with the rules stated above!

Reply:
https://github.com/guardian/cdk/pull/352 revises the linting rules